account_financial_report/report/trial_balance.py

```diff
@@ -300,6 +300,29 @@ def _compute_partner_amount(
                         ] += round(tb["amount_currency"], 2)
         return total_amount, partners_data
 
+    def _remove_accounts_at_cero(self, total_amount, show_partner_details):
+        def is_removable(d):
+            return (
+                d["initial_balance"] == 0
+                and d["credit"] == 0
+                and d["debit"] == 0
+                and d["ending_balance"] == 0
+            )
+
+        accounts_to_remove = []
+        for acc_id, ta_data in total_amount.items():
+            if is_removable(ta_data):
+                accounts_to_remove.append(acc_id)
+            elif show_partner_details:
+                partner_to_remove = []
+                for key, value in ta_data.items():
+                    if isinstance(key, int) and is_removable(value):
```

Comment:
Why do you need to do this `isinstance` thing? Always explain as comment the weird things.

Reply:
I have put a comment in the code, although it is difficult for me to explain. The image below shows the value of the total_amount variable (the main dict used for printing the report) while debugging the code in a use case. If the show_partner_details option is checked, in addition to the "initial_balance" ... "ending_balance" keys (marked in red), the same account data dictionary will also have the partner IDs as key (marked in blue) with their respective data. That's why i used isinstance method. Check the comment i have put.

![image](https://user-images.githubusercontent.com/38267832/156811846-8dc40f82-dd1f-4fa4-bd36-18bfebde6370.png)

And this is the result with this data and without my code:

![image](https://user-images.githubusercontent.com/38267832/156814049-b42d87f7-5f95-4403-be6f-5f45925069b1.png)